packages/admin-ui-docs/src/components/CodeBlockWrapper/useCopyCodeBlock.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { useState } from 'react'
+import type { MouseEventHandler, ReactNode } from 'react'
+
+import { Button, IconCopySimple } from '@vtex/admin-ui'
+
+interface ToggleCodeButtonProps {
+  children: ReactNode
+  onClick: MouseEventHandler<HTMLButtonElement>
+}
+
+const CopyCodeButton = (props: ToggleCodeButtonProps) => {
+  const { children, onClick } = props
+
+  return (
+    <Button
+      variant="neutralTertiary"
+      icon={<IconCopySimple />}
+      onClick={onClick}
+    >
+      {children}
+    </Button>
+  )
+}
+
+export function useCopyCodeBlock(code: string) {
+  const [isCodeCopied, setCodeCopied] = useState(false)
+
+  const handleCopyCode = () => {
+    if (!navigator) {
+      return
+    }
+
+    navigator.clipboard.writeText(code)
+
+    setCodeCopied(true)
+    setTimeout(() => setCodeCopied(false), 1000)
+  }
+
+  return {
+    isCodeCopied,
+    CopyCodeButton: () => (
+      <CopyCodeButton onClick={handleCopyCode}>
+        {isCodeCopied ? 'Copied' : 'Copy code'}
+      </CopyCodeButton>
```

Comment:
Maybe you can use the button direct here, what do you think?
```suggestion
      <Button
        variant="neutralTertiary"
        icon={<IconCopySimple />}
        onClick={handleCopyCode}
      >
        {isCodeCopied ? 'Copied' : 'Copy code'}
      </Button>
```